### PR TITLE
style: Ensure that precomputed pseudo-element declarations are correctly sorted.

### PR DESCRIPTION
**Pull request by emilio**
Fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1337657

r? @SimonSapin

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15459)

### CODE REVIEW

File: components/style/stylist.rs [hunk 1]
```diff
@@ -922,20 +918,21 @@ impl SelectorMap {
...
         if self.empty {
-            return
+            return vec![];
         }
 
-        let init_len = matching_rules_list.len();
+        let mut matching_rules_list = vec![];
 
+        // We need to insert important rules _after_ normal rules for this to be
+        // correct, and also to not trigger rule tree assertions.
```
Comment (emilio) [inline components/style/stylist.rs]: I guess only to be correct is enough, but... :)

**Comment by SimonSapin** [comment]
```rust
error: method `get_universal_rules` is private
   --> /home/servo/buildbot/slave/linux-dev/build/tests/unit/style/stylist.rs:117:9
    |
117 |     map.get_universal_rules(&mut decls,
    |         ^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

error: Could not compile `style_tests`.
```

**Comment by emilio** [comment]
Whoops.

@bors-servo r=SimonSapin

**Pull request merged**